tests/test_single_value_per_month.R

```diff
@@ -0,0 +1,483 @@
+library(climdex.pcic)
+library(RUnit)
+
+climdex.pcic.test.single.monthly.scalar.raw.and.csv.construction <- function() {
+  set.seed(123)
+
+  scalar_data <- runif(12, 0, 20) # One value per month
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "month", length.out = 12)
+  scalar_obj_raw <- climdexSingleMonthlyScalar.raw(
+    data = scalar_data,
+    dates = dates,
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  csv_data <- data.frame(date = as.character(dates), data = scalar_data)
+  temp_csv <- tempfile()
+  write.csv(csv_data, temp_csv, row.names = FALSE)
+
+  scalar_obj_csv <- climdexSingleMonthlyScalar.csv(
+    file = temp_csv,
+    data.column = "data",
+    date.columns = "date",
+    date.format = "%Y-%m-%d",
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  checkEquals(scalar_obj_raw@dates, scalar_obj_csv@dates, "Date mismatch between raw and CSV scalar objects.")
+  checkTrue(all.equal(scalar_obj_csv, scalar_obj_raw), msg = "Scalar_obj built from CSV is not identical to raw")
+}
+
+climdex.pcic.test.single.monthly.vector.raw.and.csv.construction <- function() {
+  set.seed(123)
+
+  primary_data <- runif(12, 0, 20) # One value per month
+  secondary_data <- runif(12, 0, 360)
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "month", length.out = 12)
+
+  vector_obj_raw <- climdexSingleMonthlyVector.raw(
+    primary = primary_data,
+    secondary = secondary_data,
+    dates = dates,
+    format = "polar",
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  csv_data <- data.frame(date = as.character(dates), primary = primary_data, secondary = secondary_data)
+  temp_csv <- tempfile()
+  write.csv(csv_data, temp_csv, row.names = FALSE)
+
+  vector_obj_csv <- climdexSingleMonthlyVector.csv(
+    file = temp_csv,
+    primary.column = "primary",
+    secondary.column = "secondary",
+    date.columns = "date",
+    date.format = "%Y-%m-%d",
+    format = "polar",
+    northern.hemisphere = TRUE,
+    calendar = "gregorian"
+  )
+
+  checkEquals(vector_obj_raw@dates, vector_obj_csv@dates, "Date mismatch between raw and CSV vector objects.")
+  checkTrue(all.equal(vector_obj_csv, vector_obj_raw), msg = "Vector_obj built from CSV is not identical to raw")
+}
+
+climdex.pcic.test.SingleMonthlyScalar.raw.missing <- function() {
+  set.seed(123)
+
+  # Single monthly value data with an NA value
+  data <- c(1:5, NA, 7:12)
+  dates <- seq(as.PCICt("2020-01-01", cal = "gregorian"), by = "month", length.out = 12)
+
+  # Create the climdexSingleMonthlyScalar object and expect it to pass without failure
+  result <- try(
+    scalar_obj <- climdexSingleMonthlyScalar.raw(data, dates, northern.hemisphere = TRUE, calendar = "gregorian"),
+    silent = TRUE
+  )
+
+  checkTrue(
+    !inherits(result, "try-error"),
+    "Function raised an error despite valid monthly data."
+  )
```

Comment:
Not sure I understand: Why wouldn't an NA value make it throw an error?

Reply:
It's possible that climate data may be incomplete for certain dates, for example, due to sensor issues. While we cannot accept data with NA dates, we do accept missing data for all our input classes.